src/util/simplify_expr.cpp

```diff
@@ -2141,24 +2147,26 @@ bool simplify_exprt::simplify_byte_update(byte_update_exprt &expr)
   // byte_extract
   if(root.id()==ID_array)
   {
-    mp_integer el_size=pointer_offset_bits(op_type.subtype(), ns);
-    if(el_size<=0 || el_size%8!=0 || val_size%8!=0)
+    auto el_size = pointer_offset_bits(op_type.subtype(), ns);
+    if(
+      !el_size.has_value() || *el_size == 0 || (*el_size) % 8 != 0 ||
+      (*val_size) % 8 != 0)
       return true;
```

Comment:
consider putting braces around `return` (preceding multi-line conditional) to improve readability